pkg/tfbridge/provider.go

```diff
@@ -74,6 +74,7 @@ type Provider struct {
 	pulumiSchema     []byte                             // the JSON-encoded Pulumi schema.
 	pulumiSchemaSpec *pschema.PackageSpec
 	memStats         memStatCollector
+	hasTypeErrors    bool
```

Comment:
I might go with turning `hasTypeErrors` into a map. Not that re-running it is too painful, but eventually when we turn it into a hard failure in `check` it will never run in Create or Update.